releasenotes/notes/lazy-load-provider-factory-bdfb3925a2514ef7.yaml

```diff
@@ -0,0 +1,22 @@
+---
+upgrade:
+  - |
+    The ``qiskit.Aer`` and ``qiskit.IBMQ`` top level attributes are now lazy
+    loaded. This means that the objects will always now exist and warnings will
+    no longer be raised on import if ``qiskit-aer`` or ``qiskit-ibmq-provider``
+    are not installed (or can't be found by Python). If you were checking for
+    the presence of ``qiskit-aer`` or ``qiskit-ibmq-provider`` using these
+    module attributes and explicitly comparing to ``None`` or looking for the
+    absence of the attribute this no longer will work because they are always
```

Comment:
Might be worth explicitly pointing out that the `ImportError` if IBMQ or Aer aren't installed will now be deferred from the `import` call to the first attempt to access.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5619/commits/4aac88394a7d4879ccb5368dec051267719290ca